### ISSUE TITLE
FIDI ERROR: Undefined array key \"data\"

### ISSUE THREAD

**Issue by MurgoloMichele** [firefly-iii/firefly-iii#5964]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I'm trying to import a .csv file, when I try to submit the data I get this error: "The submission could not be started, or failed due to an error. Please check the log files. Sorry about this :( "

If I check the logs I see this error:

```
Now submitting transaction 1/222
[2022-04-04 11:07:42] local.DEBUG: Duplicate detection method is "classic", so this method is skipped (return true).
[2022-04-04 11:07:42] local.DEBUG: Transaction #1 is unique.
[2022-04-04 11:07:42] local.DEBUG: Going to map data for this line.
[2022-04-04 11:07:42] local.DEBUG: Submitting to Firefly III: {"group_title":null,"error_if_duplicate_hash":true,"transactions":[{"type":"withdrawal","date":"","currency_id":1,"currency_code":null,"amount":"1.000000000000","description":"CANONE MENSILE CARTA DI DEBITO MESE DI RIFERIMENTO 2022/03","source_id":1,"source_name":null,"destination_id":null,"destination_name":"(no name)","payment_date":"2022-04-04 00:00:00","process_date":"2022-04-04 00:00:00","original_source":"jc5-data-import-v0.9.2","tags":[],"source_iban":null,"source_number":null,"source_bic":null,"destination_iban":null,"destination_number":null,"destination_bic":null}]}
**[2022-04-04 11:07:42] local.ERROR: Undefined array key "data" {"exception":"[object] (ErrorException(code: 0): Undefined array key \"data\" at /var/www/data-importer/vendor/firefly-iii/api-support-classes/src/Request/PostTransactionRequest.php:76)
[stacktrace]**
#0 /var/www/data-importer/vendor/firefly-iii/api-support-classes/src/Request/PostTransactionRequest.php(76): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 /var/www/data-importer/app/Services/Shared/Import/Routine/ApiSubmitter.php(246): GrumpyDictator\\FFIIIApiSupport\\Request\\PostTransactionRequest->post()
#2 /var/www/data-importer/app/Services/Shared/Import/Routine/ApiSubmitter.php(88): App\\Services\\Shared\\Import\\Routine\\ApiSubmitter->processTransaction()
#3 /var/www/data-importer/app/Services/Shared/Import/Routine/RoutineManager.php(92): App\\Services\\Shared\\Import\\Routine\\ApiSubmitter->processTransactions()
#4 /var/www/data-importer/app/Http/Controllers/Import/SubmitController.php(145): App\\Services\\Shared\\Import\\Routine\\RoutineManager->start()
#5 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): App\\Http\\Controllers\\Import\\SubmitController->start()
#6 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\\Routing\\Controller->callAction()
#7 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Route.php(262): Illuminate\\Routing\\ControllerDispatcher->dispatch()
#8 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\\Routing\\Route->runController()
#9 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Router.php(721): Illuminate\\Routing\\Route->run()
#10 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
#11 /var/www/data-importer/app/Http/Middleware/IsReadyForStep.php(54): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#12 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): App\\Http\\Middleware\\SubmitControllerMiddleware->handle()
```

### Expected behaviour

To succesfully import data from CSV

### Debug information

`Truncated from this point <----|/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\\Routing\\Controller->callAction()
#7 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Route.php(262): Illuminate\\Routing\\ControllerDispatcher->dispatch()
#8 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\\Routing\\Route->runController()
#9 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Router.php(721): Illuminate\\Routing\\Route->run()
#10 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
#11 /var/www/data-importer/app/Http/Middleware/IsReadyForStep.php(54): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#12 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): App\\Http\\Middleware\\SubmitControllerMiddleware->handle()
#13 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#14 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle()
#15 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#16 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle()
#17 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#18 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle()
#19 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#20 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest()
#21 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Session\\Middleware\\StartSession->handle()
#22 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#23 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle()
#24 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#25 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle()
#26 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#27 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Router.php(723): Illuminate\\Pipeline\\Pipeline->then()
#28 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Router.php(698): Illuminate\\Routing\\Router->runRouteWithinStack()
#29 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Router.php(662): Illuminate\\Routing\\Router->runRoute()
#30 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Routing/Router.php(651): Illuminate\\Routing\\Router->dispatchToRoute()
#31 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(167): Illuminate\\Routing\\Router->dispatch()
#32 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
#33 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#34 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#35 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle()
#36 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#37 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#38 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
#39 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#40 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#41 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#42 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
#43 /var/www/data-importer/vendor/fruitcake/laravel-cors/src/HandleCors.php(38): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#44 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fruitcake\\Cors\\HandleCors->handle()
#45 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#46 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Http\\Middleware\\TrustProxies->handle()
#47 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#48 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(142): Illuminate\\Pipeline\\Pipeline->then()
#49 /var/www/data-importer/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(111): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#50 /var/www/data-importer/public/index.php(77): Illuminate\\Foundation\\Http\\Kernel->handle()
#51 {main}
"} 
[2022-04-04 10:57:13] local.DEBUG: isReadyForCSVStep("submit")  
[2022-04-04 10:57:13] local.DEBUG: isReadyForCSVStep: Return true  
[2022-04-04 10:57:13] local.DEBUG: Now at App\Http\Controllers\Import\SubmitController::status(conv-rEEbYcD8viDf)  
[2022-04-04 10:57:13] local.DEBUG: Now in startOrFindJob(conv-rEEbYcD8viDf)  
[2022-04-04 10:57:13] local.DEBUG: Try to see if file exists for job conv-rEEbYcD8viDf.  
[2022-04-04 10:57:13] local.DEBUG: Status file exists for job conv-rEEbYcD8viDf.  
[2022-04-04 10:59:56] local.EMERGENCY: I am a EMERGENCY message.  
[2022-04-04 10:59:56] local.ALERT: I am a ALERT message.  
[2022-04-04 10:59:56] local.CRITICAL: I am a CRITICAL message.  
[2022-04-04 10:59:56] local.ERROR: I am a ERROR message.  
[2022-04-04 10:59:56] local.WARNING: I am a WARNING message.  
[2022-04-04 10:59:56] local.NOTICE: I am a NOTICE message.  
[2022-04-04 10:59:56] local.INFO: I am a INFO message.  
[2022-04-04 10:59:56] local.DEBUG: I am a DEBUG message.  `

### Steps to reproduce

1. Import .csv file
2. Set correct date format and delimiter
3. Map column values
4. Start firefly conversion job
5. Submit the data

**Comment by JC5** [comment]
Which version?

**Comment by MurgoloMichele** [comment]
FIDI:
| Scope | Version |
| --- | --- |
| FIDI | 0.9.2 |
| Build | (unknown), (unknown) |
| PHP | 8.1.4 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Using docker? | false |
| System TZ | Europe/Rome |
| Browser TZ | Europe/Rome |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | fpm-fcgi |
| BCscale | 12 |
| Trusted proxies (.env) |  |

| User info | Value |
| --- | --- |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:98.0) Gecko/20100101 Firefox/98.0 |

Firefly:
debug information generated at 2022-04-04 11:10:08 Europe/Rome for Firefly III version **5.7.0**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.0 |
| Firefly III API | 1.5.6 |
| Build | (unknown), (unknown) |
| DB version | 18 (exp. 18) |
| PHP | `8.1.4` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Rome |
| Browser TZ | Europe/Rome |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | fpm-fcgi |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql |
| Current driver | mysql |
| Trusted proxies (.env) | `` |

**Comment by JC5** [comment]
Thanks 👍

**Comment by JC5** [comment]
Fixed in 0.9.3
